### PR TITLE
Upgrading deps, running clippy fix on nightly 1.55.0

### PR DESCRIPTION
**Pull request by dessalines**

### CODE REVIEW

**Comment by Nutomic** [comment]
CI timed out on federation tests, I restarted it.

**Comment by dessalines** [comment]
@asonix I might need your help with this one, as upgrading your `background_jobs` library to `0.9.0` seems to crash the server now with the new actix and tokio versions. The crash happens when running `background_jobs_actix::create_server` which is necessary for the activitypub messages.

```
thread 'main' panicked at 'Arbiter is not running.', /home/xxx/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-rt-2.2.0/src/arbiter.rs:187:21          [15/4815]
stack backtrace:                                                                                                                                                        
h   0: std::panicking::begin_panic                                                                                                                                      
   1: actix_rt::arbiter::Arbiter::current::{{closure}}                                                                                                                  
   2: std::thread::local::LocalKey<T>::try_with                                                                                                                         
   3: std::thread::local::LocalKey<T>::with                                                                                                                             
   4: actix_rt::arbiter::Arbiter::current                                                                                                                               
   5: background_jobs_actix::create_server                                                                                                                              
   6: lemmy_apub::activity_queue::create_activity_queue                                                                                                                 
   7: lemmy_server::main::{{closure}}                                                                                                                                   
   8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll                                                                                
   9: <tokio::task::local::RunUntil<T> as core::future::future::Future>::poll::{{closure}}::{{closure}}                                                                 
  10: tokio::coop::with_budget::{{closure}}                                                                                                                             
  11: std::thread::local::LocalKey<T>::try_with                                                                                                                         
  12: std::thread::local::LocalKey<T>::with                                                                                                                             
  13: <tokio::task::local::RunUntil<T> as core::future::future::Future>::poll::{{closure}}                                                                              
  14: tokio::macros::scoped_tls::ScopedKey<T>::set                                                                                        
  15: tokio::task::local::LocalSet::with                                                                                                  
  16: <tokio::task::local::RunUntil<T> as core::future::future::Future>::poll                                                             
  17: tokio::task::local::LocalSet::run_until::{{closure}}                                                                                
  18: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll                                                  
  19: tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on::{{closure}}::{{closure}}                                              
  20: tokio::coop::with_budget::{{closure}}                                                                                               
  21: std::thread::local::LocalKey<T>::try_with                                                                                           
  22: std::thread::local::LocalKey<T>::with                                                                                               
  23: tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on::{{closure}}                                                           
  24: tokio::runtime::basic_scheduler::enter::{{closure}}         
  25: tokio::macros::scoped_tls::ScopedKey<T>::set                
  26: tokio::runtime::basic_scheduler::enter                                                                                              
  27: tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on
  28: tokio::runtime::Runtime::block_on::{{closure}}
  29: tokio::runtime::context::enter                                                
  30: tokio::runtime::handle::Handle::enter                                         
  31: tokio::runtime::Runtime::block_on                                             
  32: tokio::task::local::LocalSet::block_on                                        
  33: actix_rt::runtime::Runtime::block_on                                          
  34: actix_rt::builder::SystemRunner::block_on                                     
  35: lemmy_server::main                  
  36: core::ops::function::FnOnce::call_once                                        
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.                                                                                 
Panic in Arbiter thread. 
```

**Comment by asonix** [comment]
you can't upgraded to 0.9.0 without upgrading actix-web to the 4.0 betas. go back to the older version

**Comment by asonix** [comment]
oh wait, you did... i'll take a look

**Comment by asonix** [comment]
wait a sec, you upgraded tokio & actix-rt, but you didn't upgrade actix or actix-web, and you're using `actix_web::main` to start your runtime, which still starts an actix-rt 1 runtime

you gotta do a full upgrade or not at all

**Comment by dessalines** [comment]
Thx, I'll try this.

File: crates/utils/src/rate_limit/mod.rs [hunk 1]
```diff
@@ -190,7 +192,7 @@ where
     self.service.poll_ready(cx)
   }
 
-  fn call(&mut self, req: S::Request) -> Self::Future {
+  fn call(&mut self, req: Self::Request) -> Self::Future {
```
Comment (dessalines) [inline crates/utils/src/rate_limit/mod.rs:195]: @asonix I upgraded to actix-web `4.0.0-beta.8` but actix changed how they do transforms, and I'm pretty lost on how to get this file compiling again, I could really use your help.

[Here's their docs on Transform](https://docs.rs/actix-web/4.0.0-beta.8/actix_web/dev/trait.Transform.html)

Comment (asonix) [reply crates/utils/src/rate_limit/mod.rs:195]: &must Self turns into &Self, Request moves from an associated type to a generic on Service and Transform. It's pretty mechanical

Comment (dessalines) [reply crates/utils/src/rate_limit/mod.rs:195]: With what trait bounds? I can't do `let ip_addr = get_ip(&req.connection_info());` on a generic.

Comment (asonix) [reply crates/utils/src/rate_limit/mod.rs:195]: Instead of being generic over the Request type, only implement the trait over ServiceRequest

Comment (dessalines) [reply crates/utils/src/rate_limit/mod.rs:195]: Cool, thx!

**Comment by dessalines** [comment]
@asonix , another issue, with the newest version of `http-signature-normalization-actix`: 

https://github.com/lemmynet/lemmy/blob/upgrade_deps_4/crates/apub_receive/src/routes.rs#L78

```
error[E0277]: the trait bound `VerifyDigest<Sha256>: Transform<actix_web::scope::ScopeService, ServiceRequest>` is not satisfied
  --> crates/apub_receive/src/routes.rs:78:17
   |
78 |           .wrap(digest_verifier)
   |                 ^^^^^^^^^^^^^^^ the trait `Transform<actix_web::scope::ScopeService, ServiceRequest>` is not implemented for `VerifyDi
gest<Sha256>`
```

**Comment by asonix** [comment]
@dessalines update http-signature-normalization-actix to `0.5.0-beta.6`

File: crates/routes/src/feeds.rs [hunk 2]
```diff
@@ -124,12 +124,15 @@ async fn get_feed_data(
...
 async fn get_feed(
-  web::Path((req_type, param)): web::Path<(String, String)>,
+  req: HttpRequest,
   info: web::Query<Params>,
   context: web::Data<LemmyContext>,
 ) -> Result<HttpResponse, Error> {
   let sort_type = get_sort_type(info).map_err(ErrorBadRequest)?;
 
+  let req_type: String = req.match_info().get("type").unwrap().parse().unwrap();
+  let param: String = req.match_info().get("name").unwrap().parse().unwrap();
```
Comment (Nutomic) [inline crates/routes/src/feeds.rs:134]: Please handle this without unwrap.

Comment (dessalines) [reply crates/routes/src/feeds.rs]: K got it.

File: crates/routes/src/feeds.rs [hunk 3]
```diff
@@ -124,12 +124,15 @@ async fn get_feed_data(
...
 async fn get_feed(
-  web::Path((req_type, param)): web::Path<(String, String)>,
+  req: HttpRequest,
   info: web::Query<Params>,
   context: web::Data<LemmyContext>,
 ) -> Result<HttpResponse, Error> {
   let sort_type = get_sort_type(info).map_err(ErrorBadRequest)?;
 
+  let req_type: String = req.match_info().get("type").unwrap_or("none").parse()?;
+  let param: String = req.match_info().get("name").unwrap_or("none").parse()?;
```
Comment (Nutomic) [inline crates/routes/src/feeds.rs:134]: Hmm that seems to be the same as before. Why did you remove the web::Path at all, did they remove that?

Comment (dessalines) [reply crates/routes/src/feeds.rs:134]: Yep they removed that tuple and made it private.

Comment (Nutomic) [reply crates/routes/src/feeds.rs:134]: Damn
